airbnb/src/contexts/CalendarProvider.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { useMemo, useReducer } from 'react';
+import { compareDate } from 'utility/dateUtil';
+
+export const CalendarContext = React.createContext();
+
+function inputDateReducer(state, { type, date }) {
+  let newState = {};
+
+  switch (type) {
+    case 'INPUT_CHECK_IN':
+      newState = { ...state, checkIn: date };
```

Comment:
별도 처리가 없으면 newState 없이 바로 return 해주셔도 될 것 같네요 
```suggestion
      return { ...state, checkIn: date };
```